### PR TITLE
Add GitHub Actions workflow for publishing types

### PR DESCRIPTION
**Pull request by VictorDiniz89**

### CODE REVIEW

File: .github/workflows/publish-types.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Publish Types
+
+on:
+  release:
+    types: [created, published]
+  push:
+    branches:
+      - staging
```
Comment (chungthuang) [inline .github/workflows/publish-types.yaml:8]: We don't have a staging branch. The main branch acts as our staging.

File: .github/workflows/publish-types.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          node-version: '14'
+
+      - name: Install dependencies
+        run: npm install
+
+      - name: Publish types
+        run: cd types/generated && npm publish
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
```
Comment (chungthuang) [inline .github/workflows/publish-types.yaml:28]: I'll need to add this in the repo secret.

File: types/package.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "types-generated",
```
Comment (chungthuang) [inline types/package.json:2]: Can we rename this to something meaningful? Like devlaunchers-strapi.

File: .github/workflows/publish-types.yaml (hunk 2 above)
Comment (chungthuang) [reply .github/workflows/publish-types.yaml:28]: Added.

File: types/package.json (hunk 3 above)
Comment (VictorDiniz89) [reply types/package.json]: done

File: .github/workflows/publish-types.yaml (hunk 1 above)
Comment (VictorDiniz89) [reply .github/workflows/publish-types.yaml]: types: [created, published]
  push:
    branches:
      - main

File: .github/workflows/publish-types.yaml (hunk 2 above)
Comment (VictorDiniz89) [reply .github/workflows/publish-types.yaml:28]: ok

**Comment by VictorDiniz89** [comment]
I've done all switches you asked on the comment above

**Review by chungthuang — APPROVED** [review]

**Pull request merged**